riscv/cachesim.cc

```diff
@@ -159,6 +159,28 @@ void cache_sim_t::access(uint64_t addr, size_t bytes, bool store)
     *check_tag(addr) |= DIRTY;
 }
 
+void cache_sim_t::clean_invalidate(uint64_t addr, size_t bytes, bool clean, bool inval)
+{
+  uint64_t start_addr = addr & ~(linesz-1);
+  uint64_t end_addr = (addr + bytes + linesz-1) & ~(linesz-1);
+  uint64_t cur_addr = start_addr;
+  while (cur_addr < end_addr) {
+    uint64_t* hit_way = check_tag(cur_addr);
+    if (likely(hit_way != NULL))
+    {
+      if (*hit_way & DIRTY)
+        writebacks++;
```

Comment:
I'm not sure what "writebacks" counts, but an actual writeback is only required if a clean is required. For inval only, a proper writeback of data would not need to occur.

Reply:
OK. I'll try to fix this.